CMakeLists.txt

```diff
@@ -184,7 +184,7 @@ target_link_libraries(scylla PRIVATE
 # Force SHA1 build-id generation
 set(default_linker_flags "-Wl,--build-id=sha1")
 include(CheckLinkerFlag)
-foreach(linker "lld" "gold")
+foreach(linker "mold" "lld" "gold")
```

Comment:
Instead of an ever growing hard-coded list, let's allow the user to plug in their preferred linker via a config variable. When this variable is not provided, fall back to the hard-coded default(s).